plugins/sqlfluff-templater-dbt/sqlfluff_templater_dbt/templater.py

```diff
@@ -446,6 +447,13 @@ def make_template(in_str):
 
         node = self._find_node(fname, config)
 
+        # We have to register the connection in dbt >= 1.0.0 ourselves
+        # In previous versions, we relied on the functionality removed in https://github.com/dbt-labs/dbt-core/pull/4062
+        if DBT_VERSION_TUPLE >= (1, 0):
+            adapter = get_adapter(self.dbt_config)
+            with adapter.connection_named("master"):
+                adapter.set_relations_cache(self.dbt_manifest)
```

Comment:
> > @jpy-git I use poetry, and removed my `venv`, and even my `poetry.lock` file to get a proper clean slate, and still encountered this issue 🤷 . Maybe I discovered a case that the CI tests don't cover? 🙂
> 
> Can you open an issue with a minimum reproducible example of the commands/config that you use to setup your virtual environment please? Seems like a poetry specific thing (potentially only in the case you install from a git branch) so let's look at it separately.

@hugohjerten you'll need to run:
```
tox -e dbt100-py39 --devenv .venv
source .venv/bin/activate
```
And then install any adapters you need, e.g.:
```
pip install dbt-snowfalke
```

Reply:
> Nice job finding this! 😄
> 
> Part of me wonders given how many people are being impacted by this issue in #2113 and that the change itself is relatively small (and presumably you've verified it locally), we could potentially push just this bit through in time for the next release. 😈 (N.B. I would only advocate this in the event that we think the setup for the live postgres db connection testing is going to take too long to fit this into the next release 👼 )

@jpy-git - unfortunately the new code I've added here now means that we have to have a database connection for the dbt templater to run in dbt >= 1.0.0. Happy to get this merged in the easiest way possible, but we do at the minimum need a database connection for the tests to run.

Reply:
Would it be useful to try and find a way to make this optional? I'm thinking if the user's dbt models don't use any dbt features that require a database connection, it may be too onerous if SQLFluff requires it, especially for CI/CD environments.

Is there a way we could try and connect, but continue gracefully if we can't?

Reply:
(maybe even mock out the connection line in a fixture for the dbt tests)

Reply:
Oh, duh, my previous message was "off": I was thinking of dbt **users** always requiring a database, not the **tests**. Mocking does seem like a possibility.

Question: How valuable is it for the tests to use an actual database, versus something like mocks? In other words, if we mock the database connection, how much risk do we see that this may miss a future issue that having a real database would detect? This may depend on the kind of tests we have now, or may add in the future. (I could imagine more and more tests needing this over time, depending on what sort of issues our users report.)